src/main/java/frc/robot/commands/DefaultDriveCommand.java

```diff
@@ -27,15 +31,26 @@ public DefaultDriveCommand(DrivetrainSubsystem drivetrainSubsystem,
 
     @Override
     public void execute() {
-        // You can use `new ChassisSpeeds(...)` for robot-oriented movement instead of field-oriented movement
-        m_drivetrainSubsystem.drive(
-                ChassisSpeeds.fromFieldRelativeSpeeds(
-                        m_translationXSupplier.getAsDouble(),
-                        m_translationYSupplier.getAsDouble(),
-                        m_rotationSupplier.getAsDouble(),
-                        m_drivetrainSubsystem.getGyroscopeRotation()
-                )
-        );
+        if (m_rotationSupplier.getAsDouble() == 0 && (m_translationXSupplier.getAsDouble() != 0 || m_translationYSupplier.getAsDouble() != 0)) {
+            if (updateTarget) {
+                target = m_drivetrainSubsystem.getGyroscopeRotation();
+                updateTarget = false;
+            }
+            m_drivetrainSubsystem.drive(
+                    ChassisSpeeds.fromFieldRelativeSpeeds(
+                            m_translationXSupplier.getAsDouble(),
+                            m_translationYSupplier.getAsDouble(),
+                            -pid.calculate(m_drivetrainSubsystem.getGyroscopeRotation().minus(target).getRadians()),
```

Comment:
This code would read a little better if you wrapped the conditional logic inside another method called getRotation() then your execute would be pretty similar to before